stacks-signer/src/runloop.rs

```diff
@@ -486,9 +493,34 @@ impl<C: Coordinator> SignerRunLoop<Vec<OperationResult>, RunLoopCommand> for Run
 }
 
 /// Helper function for determining the coordinator public key given the the public keys
-fn calculate_coordinator(public_keys: &PublicKeys) -> (u32, ecdsa::PublicKey) {
-    // TODO: do some sort of VRF here to calculate the public key
-    // See: https://github.com/stacks-network/stacks-blockchain/issues/3915
-    // Mockamato just uses the first signer_id as the coordinator for now
-    (0, public_keys.signers.get(&0).cloned().unwrap())
+fn calculate_coordinator(
```

Comment:
I don't see anything wrong with this myself. :D You will def have to update the tests to not assume that the coordinator is the first signer in the list! That is why it is hanging

Reply:
Added unit tests but will need to fix the signers integ tests so the coordinator is not set statically.